crates/air_r_formatter/src/r/auxiliary/unary_expression.rs

```diff
@@ -11,17 +14,37 @@ impl FormatNodeRule<RUnaryExpression> for FormatRUnaryExpression {
         let operator = operator?;
         let argument = argument?;
 
-        write!(f, [operator.format()])?;
+        let needs_space = is_complex_anonymous_function(&operator, &argument);
 
-        if f.comments().has_comments(argument.syntax())
-            && !f.comments().is_suppressed(argument.syntax())
-        {
-            // We never break between the operator and its argument except if a
-            // newline separates them
-            write!(f, [soft_block_indent(&argument.format())])
-        } else {
-            // A unary operator always sticks to its argument
-            write!(f, [argument.format()])
-        }
+        let argument = format_with(|f| {
+            if f.comments().has_comments(argument.syntax())
+                && !f.comments().is_suppressed(argument.syntax())
+            {
+                // We never line break between the operator and its argument except if an
+                // existing comment separates them
+                write!(f, [soft_block_indent(&argument.format())])
+            } else {
+                write!(f, [argument.format()])
```

Comment:
This comment confused me at first because it mentions the behaviour in the other branch

```suggestion
                // If an existing comment separates the operator and its
                // argument we are forced to line break
                write!(f, [soft_block_indent(&argument.format())])
            } else {
                // Otherwise we never line break
                write!(f, [argument.format()])
```